src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -177,6 +225,107 @@ citus_relation_size(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * OpenConnectionToNodes opens a single connection per node
+ * for the given workerNodeList.
+ */
+static List *
+OpenConnectionToNodes(List *workerNodeList)
```

Comment:
this method should not be here I think, as a general observations I feel like we have lots of duplication around opening connections/sending queries/receiving results. We can possibly have generic functions for this purpose. (Even though we have some, we could possibly have a few more).